core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/issue/ValidationIssueSearchRequest.java

```diff
@@ -97,14 +112,28 @@ protected Expression prepareQuery(ServiceProvider context) {
 				filterByRuleIds.addAll(ruleFilter);
 			} else {
 				filterByRuleIds = newHashSet(ruleFilter);
-				
 			}
 		}
 		
 		if (filterByRuleIds != null) {
 			queryBuilder.filter(Expressions.matchAny(ValidationIssue.Fields.RULE_ID, filterByRuleIds));
 		}
 		
+		if (containsKey(OptionKey.AFFECTED_COMPONENT_ID)) {
+			Collection<String> mustComponentIds = getCollection(OptionKey.AFFECTED_COMPONENT_ID, String.class);
+			queryBuilder.filter(Expressions.matchAny(ValidationIssue.Fields.AFFECTED_COMPONENT_ID, mustComponentIds));
+		}
+		
+		if (containsKey(OptionKey.AFFECTED_COMPONENT_TYPE)) {
+			Collection<Integer> affectedComponentTypes = getCollection(OptionKey.AFFECTED_COMPONENT_TYPE, Short.class).stream().map(Integer::new).collect(Collectors.toSet());
```

Comment:
Using `Integer::valueOf` instead of `Integer::new` might mean that we hit the cached values of -128...128 and so use less objects, but it's not a big difference.